### PR TITLE
Rename files and types for clarity over energy

### PR DESCRIPTION
**Pull request by bhourahine**
Also directly expose energy_init instead of requiring signature
detection. Additionally fix a dftb_ module name for dftbp_.

### CODE REVIEW

File: prog/dftb+/lib_dftb/energytypes.F90 [hunk 1]
```diff
@@ -8,13 +8,13 @@
...
 !> Module to wrap around the different energy components in the DFTB total energy expression
-module dftbp_energies
+module dftbp_energyTypes
   use dftbp_assert
   use dftbp_accuracy
   implicit none
   private
 
-  public :: TEnergies, init
+  public :: TEnergies, energies_init
```
Comment (awvwgk) [inline prog/dftb+/lib_dftb/energytypes.F90:17]: Didn't we aim for a `TEnergies_init` naming convention with those initializers?

File: prog/dftb+/lib_dftb/energytypes.F90 [hunk 2]
```diff
@@ -8,13 +8,13 @@
 #:include 'common.fypp'
 
 !> Module to wrap around the different energy components in the DFTB total energy expression
-module dftbp_energies
+module dftbp_energyTypes
```
Comment (awvwgk) [inline prog/dftb+/lib_dftb/energytypes.F90:11]: The naming convention for modules prefers snakecase here, please avoid capital letters in module names.

File: prog/dftb+/lib_timedep/dynamics.F90 [hunk 3]
```diff
@@ -42,8 +42,8 @@ module dftbp_timeprop
   use dftbp_periodic
   use dftbp_velocityverlet
   use dftbp_nonscc
-  use dftbp_energies, only: TEnergies, init
-  use dftb_evaluateenergies
+  use dftbp_energytypes
+  use dftbp_getenergies
```
Comment (awvwgk) [inline prog/dftb+/lib_timedep/dynamics.F90:46]: Could you make the imports here explicit?

Comment (bhourahine) [reply prog/dftb+/lib_timedep/dynamics.F90]: Imports all of the public routines, but sure.

**Pull request merged**